### PR TITLE
Move using of phi arguments out of TryRemoveTrivial method

### PR DESCRIPTION
**Pull request by ptomin**
- Create unit test to reproduce uses corruption after trying to remove trivial phi
- Move using of phi arguments out of `TryRemoveTrivial` method

### CODE REVIEW

**Comment by uxmal** [comment]
I'm actively reviewing this Pavel, looking good so far. I may not be able to sign off until later tonight though. I want to test it some of the binaries in my 'zoo' but I have a prior engagement.

File: src/Decompiler/Analysis/SsaTransform.cs [hunk 1]
```diff
@@ -1264,7 +1270,7 @@ private static void GeneratePhiFunction(SsaIdentifier phi, SsaIdentifier[] sids)
             /// </summary>
             /// <param name="phi"></param>
             /// <returns></returns>
-            private SsaIdentifier TryRemoveTrivial(SsaIdentifier phi)
```
Comment (uxmal) [inline src/Decompiler/Analysis/SsaTransform.cs]: Consider documenting this function, explaining what returning true and false mean.

File: src/Decompiler/Analysis/SsaTransform.cs [hunk 2]
```diff
@@ -1262,9 +1268,13 @@ private static void GeneratePhiFunction(SsaIdentifier phi, SsaIdentifier[] sids)
...
-            /// <param name="phi"></param>
-            /// <returns></returns>
-            private SsaIdentifier TryRemoveTrivial(SsaIdentifier phi)
+            /// <param name="phi">SSA identifier of phi function</param>
+            /// <param name="sid">
+            /// Returns SSA identifier for simplified phi function if it was
+            /// trivial, otherwise returns null
+            /// </param>
+            /// <returns>true if the phi function was trivial</returns>
+            private bool TryRemoveTrivial(SsaIdentifier phi, out SsaIdentifier sid)
```
Comment (ptomin) [inline src/Decompiler/Analysis/SsaTransform.cs]: Is it enough, John?

File: src/Decompiler/Analysis/SsaTransform.cs (hunk 1 above)
Comment (ptomin) [reply src/Decompiler/Analysis/SsaTransform.cs]: I have done rebase to add documentation (see https://github.com/uxmal/reko/pull/638#discussion_r191077306).

**Pull request merged**